docs/blog/posts/gpus_lisp_machines.md

```diff
@@ -0,0 +1,115 @@
+
+## The "Better Enough"
+
+As the WebGPU optimization post noted, [WebGPU shader compilers don't do any of these optimizations](https://www.nuss-and-bolts.com/i/150233403/kernel-unrolling) and you need to unroll your loops yourself. This is likely due to game developers inserting shader compilation during frame generation so the shader compiler developers are pressured to write the *fastest* compiler they can, not the most optimal one. But there's nothing that stops us from writing our own optimizing compiler on top of [WGSL](https://www.w3.org/TR/WGSL/).
+
+That's only one part of lowering the barrier to entry to GPGPU programming, though. More important than that, the ergonomics of the GPU compute need to be much better. GPU compute should "feel" not much different to CPU programming, *but* it should have a little bit of friction on loading data into and out of the GPU, as that is the biggest bottleneck in the system. Explicitly converting an array of CPU data into a buffer of GPU data and a separate but similar set of types in the type system that you mostly can't mix together but have near identical semantic behavior within the language would go a long way there.
+
+This is an explicit objective of the Alan programming language. The focus at the moment is on getting the GPGPU integration and type system into the mostly-ergonomic, discouraging-bad-usage state, and then adding an optimizing compiler in afterwards to raise "idiomatic Alan" into "optimized WGSL."
+
+The language will work on any CPU architecture (testing on x86-64 and ARM64, was testing on RISC-V until I melted my test board), any GPU (testing on nVidia, AMD, Intel, Apple, and Broadcom GPUs), and any operating environment (testing on Windows, MacOS, Linux, and Chromium right now. Want to expand to BSD, Android, and iOS eventually), while looking similar-ish to Typescript for easier adoption. It's *already* much nicer to work with than the WebGPU API and capable of 90% of the same things.
+
+The unfortunately poor management of Symbolics doomed it, but that wouldn't have doomed the entire Lisp Machine concept. What doomed them all was an lack of desire or inability to make them more accessible to "regular" developers, increasing the supply of talent, some of whom would be hired by the existing demand that they had demonstrated, and some of whom would become new sources of demand as they came up with new and interesting ways to use it. By raising the bar on who could use it by raising the stakes so high on its use, they strangled the ecosystem before it could self-support without outside funding propping it up.
```

Comment:
s/an lack/a lack/

Reply:
I think I had `inability` first and then reordered it. Fixed!